SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialViewWorkerConnection.cpp

```diff
@@ -0,0 +1,167 @@
+#include "Interop/Connection/SpatialViewWorkerConnection.h"
+
+#include "SpatialGDKSettings.h"
+#include "SpatialView/CommandRequest.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/ConnectionHandler/SpatialOSConnectionHandler.h"
+#include "SpatialView/ViewCoordinator.h"
+
+namespace
+{
+
+SpatialGDK::ComponentData ToComponentData(FWorkerComponentData* Data)
+{
+	return SpatialGDK::ComponentData(SpatialGDK::OwningComponentDataPtr(Data->schema_type), Data->component_id);
+}
+
+SpatialGDK::ComponentUpdate ToComponentUpdate(FWorkerComponentUpdate* Update)
+{
+	return SpatialGDK::ComponentUpdate(SpatialGDK::OwningComponentUpdatePtr(Update->schema_type), Update->component_id);
+}
+
+}  // anonymous namespace
+
+void USpatialViewWorkerConnection::SetConnection(Worker_Connection* WorkerConnectionIn)
+{
+	TUniquePtr<SpatialGDK::SpatialOSConnectionHandler> Handler = MakeUnique<SpatialGDK::SpatialOSConnectionHandler>(WorkerConnectionIn);
+	Coordinator = MakeUnique<SpatialGDK::ViewCoordinator>(MoveTemp(Handler));
+}
+
+void USpatialViewWorkerConnection::FinishDestroy()
+{
+	Coordinator.Reset();
+	Super::FinishDestroy();
+}
+
+void USpatialViewWorkerConnection::DestroyConnection()
+{
+	Coordinator.Reset();
+}
+
+TArray<SpatialGDK::OpList> USpatialViewWorkerConnection::GetOpList()
+{
+	check(Coordinator.IsValid());
+	TArray<SpatialGDK::OpList> OpLists;
+	OpLists.Add(Coordinator->Advance());
+	return OpLists;
+}
+
+Worker_RequestId USpatialViewWorkerConnection::SendReserveEntityIdsRequest(uint32_t NumOfEntities)
+{
+	check(Coordinator.IsValid());
+	return Coordinator->SendReserveEntityIdsRequest(NumOfEntities);
+}
+
+Worker_RequestId USpatialViewWorkerConnection::SendCreateEntityRequest(TArray<FWorkerComponentData> Components, const Worker_EntityId* EntityId)
```

Comment:
Wonder if its worth taking TArray<FWorkerComponentData>&& so the caller is forced to give up ownership since we take schema_type below